### PR TITLE
Update minio to 2021-09-09T21-37-07Z

### PR DESCRIPTION
**Pull request by fgma**
_Motivation:_  Update minio to 2021-06-14T01-29-23Z and prepare for DSM7.
_Linked issues:_  https://github.com/SynoCommunity/spksrc/issues/4524

Depends on https://github.com/SynoCommunity/spksrc/pull/4655 being merged first to compile.

### Checklist
- [x] Build rule `all-supported` completed successfully
- [x] Package upgrade completed successfully
- [x] New installation of package completed successfully

### Checklist DSM 7
- [x] SERVICE_WIZARD_SHARE is not supported for DSM7. Shared folders must be declared in resources and must be the name of the share (without volume).
- [x] Invalid version number. DSM7 validates $(SPK_VERS)-$(SPK_REV) with regex [^\d+(\.\d+){0,5}(-\d+)?$]. So only digits and dots are allowed in SPK_VERS
- [x] SERVICE_EXE needs migration to SERVICE_COMMAND

I'm not sure about SERVICE_WIZARD_SHARE at the moment.

### CODE REVIEW

File: spk/minio/Makefile
Comment (hgy59) [inline spk/minio/Makefile:2]: SPK_VERS must not contain a `-` sign as only one is allowed.
The full version will bei `$(SPK_VERS)-$(SPK_REV)` (and SPK_REV is the only version key used by the spk repository). 
As we can ensure not to publish two versions at the same day, we could use `SPK_VERS = 2021.06.14` (and hopefully minio does not create two official releases on the same day too).

**Review by hgy59 — COMMENTED** [review]
> I'm not sure about SERVICE_WIZARD_SHARE at the moment.

I tried to implement this in #4627. But as kiwix is a new package and does not have to use the folder by the service, this was an easy job. (see new Makefile variable `SERVICE_WIZARD_SHARE_NAME`  and `mk/spksrc.service.mk`).

But there will be some effort to get the full path of the share for DSM < 7. 
Only DSM7 creates a link for shared folders under `/var/packages/{package}/shares/{share-name}`.
For DSM < 7 we must use the `synoshare` CLI tool to evaluate the full path for a shared folder by share-name.
Hopefully there is no special handling for package upgrade required.

File: spk/minio/Makefile
Comment (fgma) [reply spk/minio/Makefile]: Minio already did multiple releases the same day in the past. I guess this shouldn't be a big problem for us because the real version is still available in the cross makefile.

Fixed in https://github.com/SynoCommunity/spksrc/pull/4683/commits/7651bb27af4126305aaca5962b1aa1c15e4a042b

**Comment by fgma** [comment]
> > I'm not sure about SERVICE_WIZARD_SHARE at the moment.
> 
> I tried to implement this in #4627. But as kiwix is a new package and does not have to use the folder by the service, this was an easy job. (see new Makefile variable `SERVICE_WIZARD_SHARE_NAME` and `mk/spksrc.service.mk`).
> 
> But there will be some effort to get the full path of the share for DSM < 7.
> Only DSM7 creates a link for shared folders under `/var/packages/{package}/shares/{share-name}`.
> For DSM < 7 we must use the `synoshare` CLI tool to evaluate the full path for a shared folder by share-name.
> Hopefully there is no special handling for package upgrade required.

I'm going to wait until there is a working example merged before starting a different approach because I don't have access to a DSM7 device for testing.

File: cross/minio/Makefile
Comment (ymartin59) [inline cross/minio/Makefile:3]: I have just a proposal to avoid copy-paste mistake: generate PKG_VERS from PKG_VERS_TIMESTAMP with ':' character replacement to '-'

**Review by ymartin59 — APPROVED** [review]

File: cross/minio/Makefile
Comment (hgy59) [reply cross/minio/Makefile:3]: I have already done this but not yet pushed, as I wanted to redesign shared folders too.
So I will do an intermedeate commit.

**Review by ymartin59 — APPROVED** [review]

**Comment by fgma** [comment]
Any news on how to continue concerning SERVICE_WIZARD_SHARE?

**Comment by hgy59** [comment]
> 
> 
> Any news on how to continue concerning SERVICE_WIZARD_SHARE?

see  #4748 (almost ready to merge)

**Comment by th0ma7** [comment]
something is wrong with this PR: there are tons of non-related master branch changes pulled into this PR, it needs cleaning-up.
You probably updated your branch directly from spksrc instead of updating your master branch from spksrc and then updating your branch through your master branch locally, resulting in this.  You may need to create a new branch updated against master, then cherry-pick all your commits, and push a new PR using that.

Here's how I normally update my PR branch:

1. Update your master branch first:
```
git checkout master
git fetch upstream
git rebase upstream/master
git push --force origin master
```

2. Then update your local branch associated to your PR:
```
git checkout <my-local-branch>
git rebase master
git push --force
```

**Comment by fgma** [comment]
@th0ma7 Thanks for your help. I've already noticed I messed something up but did not have the time to clean it up yet.

**Comment by fgma** [comment]
@th0ma7 Rebasing it again fixed the problem. I hope to get the rest done soon.

**Comment by fgma** [comment]
@th0ma7 Seems like https://github.com/SynoCommunity/spksrc/commit/a442e16582503d6562605aedd7a2aa6b9d4e7165 broke the minio build because POST_EXTRACT_TARGET is not called anymore. Can you have a look at this?

**Comment by fgma** [comment]
I've upgraded minio to the latest release and changed the wizard similar to transmission. Currently the following issues are open:

1. The build fails because of https://github.com/SynoCommunity/spksrc/commit/a442e16582503d6562605aedd7a2aa6b9d4e7165. When there is a fix merged I will rebase my branch to clean it up.
2. The wizard does not show user/password fields on upgrade which I don't understand why
3. Do I need a special 7.0 subfolder for the wizard or can I use a single one for 6.1 and 7.0?

**Comment by th0ma7** [comment]
@fgma I gave it a shot on master, on my new PR  #4854 (contains a few parallel build fixes) and from your branch on your clone and I can't reproduce your issue.

I've simply added a trace in `cross/minio/Makefile` such as the following and it always printed...
```
# rename extracted folder to make imports work when used as GOPATH
minio_post_extract:
        cd $(EXTRACT_PATH) && [ -d $(PKG_DIR) ] && mv $(PKG_DIR) $(PKG_NAME)
        @$(MSG) ~~~~~~~~~~~~~~~~~~~~ post extract from minio Makefile
```
As such, `POST_EXTRACT_TARGET` is indeed being called...

**Comment by fgma** [comment]
I test it against a clean checkout of master:

```git clone https://github.com/SynoCommunity/spksrc
docker pull ghcr.io/synocommunity/spksrc
docker run -it -v /tmp/spksrc:/spksrc ghcr.io/synocommunity/spksrc /bin/bash
make setup
cd spk/minio
make arch-armada38x
```

This results in the following log:

```
===>  Installing for busybox
find /spksrc/spk/minio/work-armada38x-7.0/install//var/packages/minio/target/ \! -type d -printf '%P\n' | sort > /spksrc/spk/minio/work-armada38x-7.0/busybox.plist.tmp
install -m 755 -d /spksrc/spk/minio/work-armada38x-7.0/install/var/packages/minio/target/bin
install -m 755 /spksrc/spk/minio/work-armada38x-7.0/busybox-1.26.2/busybox /spksrc/spk/minio/work-armada38x-7.0/install/var/packages/minio/target/bin
find /spksrc/spk/minio/work-armada38x-7.0/install//var/packages/minio/target/ \! -type d -printf '%P\n' | sort | \
  diff /spksrc/spk/minio/work-armada38x-7.0/busybox.plist.tmp -  | grep '>' | cut -d' ' -f2- > /spksrc/spk/minio/work-armada38x-7.0/busybox.plist
# Generate busybox.plist.auto for newly added files (diff against .tmp)
comm -3 /spksrc/spk/minio/work-armada38x-7.0/busybox.plist.tmp /spksrc/spk/minio/work-armada38x-7.0/busybox.plist > /spksrc/spk/minio/work-armada38x-7.0/busybox.plist.auto
make[3]: Leaving directory '/spksrc/cross/busybox'
make[3]: Entering directory '/spksrc/cross/minio'
===>  Verifying files for minio
===>  Downloading files for minio
===>  Extracting for minio
===>  Processing dependencies of minio
===>  Patching for minio
===>  Compiling for minio
===>  Configuring for minio
===>  Installing for minio
===>    Checking sha1sum of file minio-RELEASE.2020-11-25T22-36-25Z.tar.gz
===>    File minio-RELEASE.2020-11-25T22-36-25Z.tar.gz already downloaded
make[4]: Entering directory '/spksrc/native/go'
make[4]: Nothing to be done for 'default'.
make[4]: Leaving directory '/spksrc/native/go'
tar -xzpf /spksrc/cross/minio/../../distrib/minio-RELEASE.2020-11-25T22-36-25Z.tar.gz -C /spksrc/spk/minio/work-armada38x-7.0/src/github.com/minio 
===>    Checking sha256sum of file minio-RELEASE.2020-11-25T22-36-25Z.tar.gz
===>    Checking md5sum of file minio-RELEASE.2020-11-25T22-36-25Z.tar.gz
/spksrc/cross/minio/../../distrib/minio-RELEASE.2020-11-25T22-36-25Z.tar.gz
===>  - Compile with go build
cd /spksrc/spk/minio/work-armada38x-7.0/src/github.com/minio/minio && env GO111MODULE=on GOARM=7 GOPATH=/spksrc/cross/minio/../../distrib/go CGO_ENABLED=0 PATH=/spksrc/spk/minio/work-armada38x-7.0/../../../native/go/work-native/go/bin/:$PATH GOARCH=arm GOOS=linux PKG_CONFIG_LIBDIR=/spksrc/spk/minio/work-armada38x-7.0/install//var/packages/minio/target/lib/pkgconfig WORK_DIR=/spksrc/spk/minio/work-armada38x-7.0 INSTALL_PREFIX=/var/packages/minio/target TC=syno-armada38x-7.0  LD="/spksrc/toolchain/syno-armada38x-7.0/work/arm-unknown-linux-gnueabi/bin/arm-unknown-linux-gnueabi-ld" LDSHARED="/spksrc/toolchain/syno-armada38x-7.0/work/arm-unknown-linux-gnueabi/bin/arm-unknown-linux-gnueabi-gcc -shared" CPP="/spksrc/toolchain/syno-armada38x-7.0/work/arm-unknown-linux-gnueabi/bin/arm-unknown-linux-gnueabi-cpp" NM="/spksrc/toolchain/syno-armada38x-7.0/work/arm-unknown-linux-gnueabi/bin/arm-unknown-linux-gnueabi-nm" CC="/spksrc/toolchain/syno-armada38x-7.0/work/arm-unknown-linux-gnueabi/bin/arm-unknown-linux-gnueabi-gcc" AS="/spksrc/toolchain/syno-armada38x-7.0/work/arm-unknown-linux-gnueabi/bin/arm-unknown-linux-gnueabi-as" RANLIB="/spksrc/toolchain/syno-armada38x-7.0/work/arm-unknown-linux-gnueabi/bin/arm-unknown-linux-gnueabi-ranlib" CXX="/spksrc/toolchain/syno-armada38x-7.0/work/arm-unknown-linux-gnueabi/bin/arm-unknown-linux-gnueabi-g++" AR="/spksrc/toolchain/syno-armada38x-7.0/work/arm-unknown-linux-gnueabi/bin/arm-unknown-linux-gnueabi-ar" STRIP="/spksrc/toolchain/syno-armada38x-7.0/work/arm-unknown-linux-gnueabi/bin/arm-unknown-linux-gnueabi-strip" OBJDUMP="/spksrc/toolchain/syno-armada38x-7.0/work/arm-unknown-linux-gnueabi/bin/arm-unknown-linux-gnueabi-objdump" READELF="/spksrc/toolchain/syno-armada38x-7.0/work/arm-unknown-linux-gnueabi/bin/arm-unknown-linux-gnueabi-readelf" CFLAGS="-I/spksrc/toolchain/syno-armada38x-7.0/work/arm-unknown-linux-gnueabi/arm-unknown-linux-gnueabi/sysroot/usr/include -D__ARM_PCS_VFP=1 -I/spksrc/spk/minio/work-armada38x-7.0/install//var/packages/minio/target/include " CPPFLAGS="-I/spksrc/toolchain/syno-armada38x-7.0/work/arm-unknown-linux-gnueabi/arm-unknown-linux-gnueabi/sysroot/usr/include -D__ARM_PCS_VFP=1 -I/spksrc/spk/minio/work-armada38x-7.0/install//var/packages/minio/target/include " CXXFLAGS="-I/spksrc/toolchain/syno-armada38x-7.0/work/arm-unknown-linux-gnueabi/arm-unknown-linux-gnueabi/sysroot/usr/include -D__ARM_PCS_VFP=1 -I/spksrc/spk/minio/work-armada38x-7.0/install//var/packages/minio/target/include " LDFLAGS="-L/spksrc/toolchain/syno-armada38x-7.0/work/arm-unknown-linux-gnueabi/arm-unknown-linux-gnueabi/sysroot/lib -D__ARM_PCS_VFP=1 -L/spksrc/spk/minio/work-armada38x-7.0/install//var/packages/minio/target/lib -Wl,--rpath-link,/spksrc/spk/minio/work-armada38x-7.0/install//var/packages/minio/target/lib -Wl,--rpath,/var/packages/minio/target/lib " go build -o /spksrc/spk/minio/work-armada38x-7.0/src/github.com/minio/minio/minio -tags kqueue -ldflags "-s -w -X github.com/minio/minio/cmd.Version=2020-11-25T22-36-25Z -X github.com/minio/minio/cmd.ReleaseTag=2020-11-25T22-36-25Z"
/bin/sh: 1: cd: can't cd to /spksrc/spk/minio/work-armada38x-7.0/src/github.com/minio/minio
make[3]: *** [../../mk/spksrc.cross-go.mk:50: go_build_target] Error 2
make[3]: *** Waiting for unfinished jobs....
make[3]: Leaving directory '/spksrc/cross/minio'
make[2]: *** [../../mk/spksrc.depend.mk:51: depend_target] Error 2
make[2]: Leaving directory '/spksrc/spk/minio'
make[1]: Leaving directory '/spksrc/spk/minio'
```

After adding the log message as suggested nothing changes:

```
===>  Downloading files for minio
===>  Verifying files for minio
===>  Processing dependencies of minio
===>  Extracting for minio
===>  Patching for minio
===>  Configuring for minio
===>  Compiling for minio
===>  Installing for minio
===>    File minio-RELEASE.2020-11-25T22-36-25Z.tar.gz already downloaded
===>    Checking sha1sum of file minio-RELEASE.2020-11-25T22-36-25Z.tar.gz
make[4]: Entering directory '/spksrc/native/go'
===>    Checking sha256sum of file minio-RELEASE.2020-11-25T22-36-25Z.tar.gz
make[4]: Nothing to be done for 'default'.
make[4]: Leaving directory '/spksrc/native/go'
tar -xzpf /spksrc/cross/minio/../../distrib/minio-RELEASE.2020-11-25T22-36-25Z.tar.gz -C /spksrc/spk/minio/work-armada38x-7.0/src/github.com/minio 
===>    Checking md5sum of file minio-RELEASE.2020-11-25T22-36-25Z.tar.gz
/spksrc/cross/minio/../../distrib/minio-RELEASE.2020-11-25T22-36-25Z.tar.gz
===>  - Compile with go build
cd /spksrc/spk/minio/work-armada38x-7.0/src/github.com/minio/minio && env GO111MODULE=on GOARM=7 GOPATH=/spksrc/cross/minio/../../distrib/go CGO_ENABLED=0 PATH=/spksrc/spk/minio/work-armada38x-7.0/../../../native/go/work-native/go/bin/:$PATH GOARCH=arm GOOS=linux PKG_CONFIG_LIBDIR=/spksrc/spk/minio/work-armada38x-7.0/install//var/packages/minio/target/lib/pkgconfig WORK_DIR=/spksrc/spk/minio/work-armada38x-7.0 INSTALL_PREFIX=/var/packages/minio/target TC=syno-armada38x-7.0  LD="/spksrc/toolchain/syno-armada38x-7.0/work/arm-unknown-linux-gnueabi/bin/arm-unknown-linux-gnueabi-ld" LDSHARED="/spksrc/toolchain/syno-armada38x-7.0/work/arm-unknown-linux-gnueabi/bin/arm-unknown-linux-gnueabi-gcc -shared" CPP="/spksrc/toolchain/syno-armada38x-7.0/work/arm-unknown-linux-gnueabi/bin/arm-unknown-linux-gnueabi-cpp" NM="/spksrc/toolchain/syno-armada38x-7.0/work/arm-unknown-linux-gnueabi/bin/arm-unknown-linux-gnueabi-nm" CC="/spksrc/toolchain/syno-armada38x-7.0/work/arm-unknown-linux-gnueabi/bin/arm-unknown-linux-gnueabi-gcc" AS="/spksrc/toolchain/syno-armada38x-7.0/work/arm-unknown-linux-gnueabi/bin/arm-unknown-linux-gnueabi-as" RANLIB="/spksrc/toolchain/syno-armada38x-7.0/work/arm-unknown-linux-gnueabi/bin/arm-unknown-linux-gnueabi-ranlib" CXX="/spksrc/toolchain/syno-armada38x-7.0/work/arm-unknown-linux-gnueabi/bin/arm-unknown-linux-gnueabi-g++" AR="/spksrc/toolchain/syno-armada38x-7.0/work/arm-unknown-linux-gnueabi/bin/arm-unknown-linux-gnueabi-ar" STRIP="/spksrc/toolchain/syno-armada38x-7.0/work/arm-unknown-linux-gnueabi/bin/arm-unknown-linux-gnueabi-strip" OBJDUMP="/spksrc/toolchain/syno-armada38x-7.0/work/arm-unknown-linux-gnueabi/bin/arm-unknown-linux-gnueabi-objdump" READELF="/spksrc/toolchain/syno-armada38x-7.0/work/arm-unknown-linux-gnueabi/bin/arm-unknown-linux-gnueabi-readelf" CFLAGS="-I/spksrc/toolchain/syno-armada38x-7.0/work/arm-unknown-linux-gnueabi/arm-unknown-linux-gnueabi/sysroot/usr/include -D__ARM_PCS_VFP=1 -I/spksrc/spk/minio/work-armada38x-7.0/install//var/packages/minio/target/include " CPPFLAGS="-I/spksrc/toolchain/syno-armada38x-7.0/work/arm-unknown-linux-gnueabi/arm-unknown-linux-gnueabi/sysroot/usr/include -D__ARM_PCS_VFP=1 -I/spksrc/spk/minio/work-armada38x-7.0/install//var/packages/minio/target/include " CXXFLAGS="-I/spksrc/toolchain/syno-armada38x-7.0/work/arm-unknown-linux-gnueabi/arm-unknown-linux-gnueabi/sysroot/usr/include -D__ARM_PCS_VFP=1 -I/spksrc/spk/minio/work-armada38x-7.0/install//var/packages/minio/target/include " LDFLAGS="-L/spksrc/toolchain/syno-armada38x-7.0/work/arm-unknown-linux-gnueabi/arm-unknown-linux-gnueabi/sysroot/lib -D__ARM_PCS_VFP=1 -L/spksrc/spk/minio/work-armada38x-7.0/install//var/packages/minio/target/lib -Wl,--rpath-link,/spksrc/spk/minio/work-armada38x-7.0/install//var/packages/minio/target/lib -Wl,--rpath,/var/packages/minio/target/lib " go build -o /spksrc/spk/minio/work-armada38x-7.0/src/github.com/minio/minio/minio -tags kqueue -ldflags "-s -w -X github.com/minio/minio/cmd.Version=2020-11-25T22-36-25Z -X github.com/minio/minio/cmd.ReleaseTag=2020-11-25T22-36-25Z"
/bin/sh: 1: cd: can't cd to /spksrc/spk/minio/work-armada38x-7.0/src/github.com/minio/minio
make[3]: *** [../../mk/spksrc.cross-go.mk:50: go_build_target] Error 2
make[3]: *** Waiting for unfinished jobs....
make[3]: Leaving directory '/spksrc/cross/minio'
make[2]: *** [../../mk/spksrc.depend.mk:51: depend_target] Error 2
make[2]: Leaving directory '/spksrc/spk/minio'
make[1]: Leaving directory '/spksrc/spk/minio'
```

**Comment by th0ma7** [comment]
@fgma I think I've finally nailed it... I recall hitting a similar problem previously.
I was able to build your PR by copying over the changes from mine successfully.
Can you give it a shot and confirm this works out for you as well (and if so, comment it on #4854)

**Comment by fgma** [comment]
I've upgraded minio to the latest release. Currently the following issues are open:

1. The wizard does not show user/password fields on upgrade which I don't understand why
2. Do I need a special 7.0 subfolder for the wizard or can I use a single one for 6.1 and 7.0?

Can anyone provide some feedback on this?

**Comment by fgma** [comment]
@hgy59 @ymartin59 Would be great if I could get some feedback so we can get this PR done soon. No problem if your lacking the time at the moment I just want make sure this PR is not forgotten.

**Comment by th0ma7** [comment]
>     1. The wizard does not show user/password fields on upgrade which I don't understand why

By experience that is due to an error somewhere in your code, often a `,` or `[` `]` type-thing... I used to remove everything and re-add one-by-one to validate it really is ok for the DSM (long and painful process).  Perhaps @hgy59 or someone else has a cheet-sheet command that allows testing that easily.

**Comment by hgy59** [comment]
For me the upgrade does not work at all 
grafik

The same spk works for a fresh installation.
DSM 7

File: spk/minio/src/wizard/install_uifile
Comment (hgy59) [inline spk/minio/src/wizard/install_uifile:49]: you must escape the `\` to fix the upgrade wizard: to get `\\` you must wirte `\\\\`
```suggestion
                                "expr": "/^[\\\\w _-]+$/",
```

File: spk/minio/src/wizard/upgrade_uifile.sh
Comment (hgy59) [inline spk/minio/src/wizard/upgrade_uifile.sh:61]: you must escape the `\` to fix the upgrade wizard: to get `\\` you must write `\\\\`
```suggestion
                                "expr": "/^[\\\\w _-]+$/",
```

**Comment by hgy59** [comment]
> By experience that is due to an error somewhere in your code, often a `,` or `[` `]` type-thing... I used to remove everything and re-add one-by-one to validate it really is ok for the DSM (long and painful process). Perhaps @hgy59 or someone else has a cheet-sheet command that allows testing that easily.

The log file `/var/log/synoscgi.log` contains the error message:

```
2021-09-21T00:14:12+02:00 ##### synoscgi_SYNO.Core.Package.Installation_1_upload[8804]: pkginfo.cpp:1430 Failed to read wizard content from file, pkg=[minio] path=[/var/run/synopkgs/synopkgwizard.log.wSn0ku] content=[[
  {
    "step_title": "MinIO configuration",
    "items": [
      {
        "type": "combobox",
        "desc": "Please select a volume to use for the data folder",
        "subitems": [
          {
            "key": "wizard_data_volume",
            "desc": "Volume name",
            "defaultValue": "/volume1",
            "displayField": "display_name",
            "valueField": "volume_path",
            "editable": false,
            "mode": "remote",
            "api_store": {
              "api": "SYNO.Core.Storage.Volume",
              "method": "list",
              "version": 1,
              "baseParams": {
                "limit": -1,
                "offset": 0,
                "location": "internal"
              },
              "root": "volumes",
              "idProperty": "volume_path",
              "fields": [
                "display_name",
                "volume_path"
              ]
            },
            "validator": {
              "fn": "{console.log(arguments);return true;}"
            }
          }
        ]
      },
      {
        "type": "textfield",
        "desc": "Data shared folder (using the volume chosen above)",
        "subitems": [
          {
            "key": "wizard_data_directory",
            "desc": "Data shared folder",
            "defaultValue": "minio",
            "validator": {
              "allowBlank": false,
              "regex": {
                "expr": "/^[\w _-]+$/",
                "errorText": "Subdirectories are not supported."
              }
            }
          }
        ]
      },
      {
        "type": "textfield",
        "subitems": [
          {
            "key": "wizard_root_user",
            "desc": "MinIO root user",
            "defaultValue": "root",
            "validator": {
              "allowBlank": false,
              "minLength": 3,
              "regex": {
                "expr": "/^[^<>:*/?\"|]*$/",
                "errorText": "Not allowed character in username"
              }
            }
          }
        ]
      },
      {
        "type": "textfield",
        "subitems": [
          {
            "key": "wizard_root_password",
            "desc": "MinIO root password",
            "defaultValue": "password",
            "validator": {
              "allowBlank": false,
              "minLength": 8,
              "regex": {
                "expr": "/^[^\"|]*$/",
                "errorText": "Not allowed character in password"
              }
            }
          }
        ]
      }
    ]
  },
  {
    "step_title": "DSM Permissions",
    "items": [
      {
        "desc": "Please read <a target=\"_blank\" href=\"https://github.com/SynoCommunity/spksrc/wiki/Permission-Management\">Permission Management</a> for details."
      }
    ]
  }
]
] msg=[No such file or directory]
2021-09-21T00:14:12+02:00 ##### synoscgi_SYNO.Core.Package.Installation_1_upload[8804]: pkgtool.cpp:1384 Failed to load wizard, pkg=[minio]
```

**Review by hgy59 — APPROVED** [review]
LGTM
i will use this install wizard as template for other packages.

**Comment by fgma** [comment]
Thanks a lot for you fast help! I tried to get the update ready but messed something up during rebase because I wanted to build against the new go version. This should be fixed now. If there are no issues left feel free to merge this anytime you like. Until then I will updated this PR with the latest minio release.

File: cross/minio/Makefile
Comment (hgy59) [inline cross/minio/Makefile:2]: i suppose you wanted to define `PKG_VERS = 2021-09-18T18:09:59Z` or the following subst is obsolete...

Comment (fgma) [reply cross/minio/Makefile]: Fixed in https://github.com/SynoCommunity/spksrc/pull/4683/commits/cdf18f1267705853e6b7608837ba0e7dd5bcb568

File: spk/minio/Makefile
Comment (hgy59) [inline spk/minio/Makefile:16]: due to the license change we might need to include a license file with `LICENSE_FILE = ...` to include the license in the installation wizard.

Comment (fgma) [reply spk/minio/Makefile:17]: Fixed in https://github.com/SynoCommunity/spksrc/pull/4683/commits/3500ca31f987db7386b4cde11e9fc347cdd727a5

**Pull request merged**
